channels/pkg/channels/addons_test.go

```diff
@@ -0,0 +1,154 @@
+			Expected:          false,
+		},
+		{
+			Input: api.AddonSpec{
+				KubernetesVersion: ">=1.6.0",
+			},
+			KubernetesVersion: "1.5.9",
+			Expected:          false,
+		},
+		{
+			Input: api.AddonSpec{
+				KubernetesVersion: ">=1.4.0 <1.6.0",
+			},
+			KubernetesVersion: "1.5.9",
+			Expected:          true,
+		},
+		{
+			Input: api.AddonSpec{
+				KubernetesVersion: ">=1.4.0 <1.6.0",
+			},
+			KubernetesVersion: "1.6.0",
+			Expected:          false,
+		},
+	}
+	for _, g := range grid {
+		k8sVersion := semver.MustParse(g.KubernetesVersion)
+		addon := &Addon{
+			Spec: &g.Input,
+		}
+		actual := addon.matches(k8sVersion)
+		if actual != g.Expected {
+			t.Errorf("unexpected result from %v, %s.  got %v", g.Input.KubernetesVersion, g.KubernetesVersion, actual)
+		}
+	}
+}
+
+func Test_Replacement(t *testing.T) {
+	grid := []struct {
+		Old      *ChannelVersion
+		New      *ChannelVersion
+		Replaces bool
+	}{
+		// With no id, update iff newer semver
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: ""},
+			New:      &ChannelVersion{Version: s("1.0.0"), Id: ""},
+			Replaces: false,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: ""},
+			New:      &ChannelVersion{Version: s("1.0.1"), Id: ""},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.1"), Id: ""},
+			New:      &ChannelVersion{Version: s("1.0.0"), Id: ""},
+			Replaces: false,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.1.0"), Id: ""},
+			New:      &ChannelVersion{Version: s("1.1.1"), Id: ""},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.1.1"), Id: ""},
+			New:      &ChannelVersion{Version: s("1.1.0"), Id: ""},
+			Replaces: false,
+		},
+
+		// With id, update if different id and same version, otherwise follow semver
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			New:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			Replaces: false,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			New:      &ChannelVersion{Version: s("1.0.0"), Id: "b"},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "b"},
+			New:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			New:      &ChannelVersion{Version: s("1.0.1"), Id: "a"},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			New:      &ChannelVersion{Version: s("1.0.1"), Id: "a"},
+			Replaces: true,
+		},
+		{
+			Old:      &ChannelVersion{Version: s("1.0.0"), Id: "a"},
+			New:      &ChannelVersion{Version: s("1.0.1"), Id: "a"},
+			Replaces: true,
+		},
+	}
+	for _, g := range grid {
+		actual := g.New.replaces(g.Old)
+		if actual != g.Replaces {
+			t.Errorf("unexpected result from %v -> %v, expect %t.  actual %v", g.Old, g.New, g.Replaces, actual)
+		}
+	}
+}
+
+func s(v string) *string {
```

Comment:
Convenience functions! I know you love these, but it creates a questionable layer of complexity..

Whats wrong with just using idiomatic `*` and `&`?